check_powerdns_rec.py

```
@@ -188,86 +290,56 @@ def calc_avgps(_data_old, _data_new):
-
-        data_new = parse_pdns(stdout)
-        data_new['epoch'] = int(time.time())
-
-        data_old = data_new.copy()
-        data_old['epoch'] -= 1
-
-        (data_avg, queries) = calc_avgps(data_old, data_new)
-    else:
-        try:
-            cli = [pdns_tool]
-            if args.socket_dir:
-                cli.append('--socket-dir=%s' % args.socket_dir)
-            if args.config_name != '':
-                cli.append('--config-name=%s' % args.config_name)
-            cli.append('get-all')
-
-            MyOut = subprocess.Popen(cli, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-            stdout, stderr = MyOut.communicate()
-            if MyOut.returncode != 0:
-                monitor.set_message(stdout)
-                monitor.report()
-        except OSError:
-            monitor.set_message("Control command '%s' not found." % pdns_tool)
-            monitor.report()
-
-        # noinspection PyUnboundLocalVariable
-        data_new = parse_pdns(stdout)
-        data_new['epoch'] = int(time.time())
-
-        filename = get_fname(args.scratch, args.config_name)
-        data_old = load_measurement(filename)
+    try:
+        args = parse_args()
 
+        if args.test:
+            data_old, data_new = fake_statistics()
+        else:
+            if args.api_host:
+                pdns = PowerDnsApi(args.api_host, args.api_port, args.api_key)
+                result = pdns.statistics()
+                data_new = parse2_pdns(result)
+            else:
+                pdns = PowerDnsCtrlTool(args.socket_dir, args.config_name)
+                result = pdns.get_all()
+                data_new = parse_pdns(result)
+            # noinspection PyUnboundLocalVariable
+            data_new['epoch'] = int(time.time())
+            filename = get_fname(args.scratch, args.config_name)
+            data_old = load_measurement(filename)
+            if len(data_new) > 1:
```

I think we're checking for `> 1` because `data_new['epoch']` is always set. I think the data provider method (`parse_pdns` in this case) should throw an exception in case the data isn't valid and usable.